File: app/src/main/java/com/scatl/uestcbbs/module/board/adapter/ForumListRightAdapter.java
package com.scatl.uestcbbs.module.board.adapter;

import android.content.Context;
import android.util.Log;
import android.widget.GridView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.ForumListBean;
import com.scatl.uestcbbs.util.SharePrefUtil;

import java.util.List;

/**
 * author: sca_tl
 * description:
 * date: 2019/8/4 16:31
 */
public class ForumListRightAdapter extends BaseQuickAdapter<ForumListBean.ListBean, BaseViewHolder> {

    public ForumListRightAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, ForumListBean.ListBean item) {
        helper.setText(R.id.forum_list_right_title, item.board_category_name);
        ForumListGridViewAdapter forumListGridViewAdapter = new ForumListGridViewAdapter(mContext, getData().get(helper.getLayoutPosition()).board_list);
        GridView gridView = helper.getView(R.id.forum_list_right_gridview);
        gridView.setNumColumns(SharePrefUtil.getBoardListColumns(mContext));
        gridView.setAdapter(forumListGridViewAdapter);
        gridView.requestFocus();
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/board/adapter/ForumListRightAdapter.kt
package com.scatl.uestcbbs.module.board.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemForumListRightBinding
import com.scatl.uestcbbs.entity.ForumListBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.SharePrefUtil

/**
 * Created by sca_tl at 2023/6/9 15:34
 */
class ForumListRightAdapter: PreloadAdapter<ForumListBean.ListBean, ItemForumListRightBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemForumListRightBinding {
        return ItemForumListRightBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemForumListRightBinding>, position: Int, item: ForumListBean.ListBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.name.text = item.board_category_name
        holder.binding.gridview.apply {
            numColumns = SharePrefUtil.getBoardListColumns(context)
            adapter = ForumListGridViewAdapter(context, items[holder.layoutPosition].board_list)
            requestFocus()
        }
    }
}